nvflare/private/fed/server/job_runner.py

```diff
@@ -215,7 +215,16 @@ def _job_complete_process(self, fl_ctx: FLContext):
                     with self.lock:
                         job = self.running_jobs.get(run_number)
                         if job:
-                            if run_number in engine.execution_exception_run_number:
+                            if run_number in engine.execution_exception_run_processes:
+                                run_process = engine.execution_exception_run_processes[run_number]
+                                # stop client run
+                                client_sites = run_process.get(RunProcessKey.PARTICIPANTS)
+
+                                replies = self.abort_client_run(engine, run_number, client_sites, fl_ctx)
+                                if not replies:
+                                    self.log_error(
+                                        fl_ctx, f"Failed to send abort command to clients for run: {run_number}"
+                                    )
```

Comment:
Why do we need this? When the ServerRunner end the job running, it already sent the ReservedTopic.END_RUN to all the running clients to end the client run. We should not need this any more.

Reply:
Checked with Yuan-ting on this. This logic is to handle the case the server app did't actually start due to the config error case. We could not use the event handling to notify client to abort run yet. The handle logic is correct.